FreeSurfer/FreeSurferPipeline.sh

```diff
@@ -34,15 +34,204 @@
+                then
+                    newargs+=(--flair "${origargs[i + 1]}")
+                    #skip the boolean value, we took care of it
+                    i=$((i + 1))
+                else
+                    newargs+=(--flair=TRUE)
+                    changeargs=1
+                fi
+                ;;
+            (--existing-subject)
+                #same logic
+                if ((i + 1 < ${#origargs[@]})) && (opts_StringToBool "${origargs[i + 1]}" &> /dev/null)
+                then
+                    newargs+=(--existing-subject "${origargs[i + 1]}")
+                    i=$((i + 1))
+                else
+                    newargs+=(--existing-subject=TRUE)
+                    changeargs=1
+                fi
+                ;;
+            (--no-conf2hires)
+                #this doesn't match a new argument, so we can just replace it
+                newargs+=(--conf2hires=FALSE)
+                changeargs=1
+                ;;
+            (--extra-reconall-arg=*)
+                #repeatable options aren't yet a thing in newopts (indirect assignment to arrays seems to need eval)
+                #figure out whether these extra arguments could have a better syntax (if whitespace is supported, probably not)
+                extra_reconall_args_manual+=("${origargs[i]#*=}")
+                changeargs=1
+                ;;
+            (--extra-reconall-arg)
+                #also support --extra-reconall-arg foo, for fewer surprises
+                if ((i + 1 >= ${#origargs[@]}))
+                then
+                    log_Err_Abort "--extra-reconall-arg requires an argument"
+                fi
+                extra_reconall_args_manual+=("${origargs[i + 1]#*=}")
+                #skip the next argument, we took care of it
+                i=$((i + 1))
+                changeargs=1
+                ;;
+            (*)
+                #copy anything unrecognized
+                newargs+=("${origargs[i]}")
+                ;;
+        esac
+    done
+    if ((changeargs))
+    then
+        echo "original arguments: $*"
+        set -- "${newargs[@]}"
+        echo "new arguments: $*"
+        echo "extra recon-all arguments: ${extra_reconall_args_manual[*]+"${extra_reconall_args_manual[*]}"}"
+    fi
+fi
+
+#description to use in usage - syntax of parameters is now explained automatically
+opts_SetScriptDescription "Runs the FreeSurfer HCP pipline on data processed by prefresurfer"
+
+# Show usage information
+opts_AddMandatory '--subject' 'SubjectID' 'subject' "Subject ID (required)  Used with --path input to create full path to root  directory for all outputs generated as path/subject"
+
+opts_AddOptional  '--subjectDIR' 'SubjectDIR' 'subject' 'path to subject directory required, unless --existing-subject is set' " " "--subject-dir"
+	 
+opts_AddOptional '--t1' 'T1wImage' "T1" 'path to T1w image required, unless --existing-subject is set' " " "--t1w-image"
```

Comment:
We should probably swap these around so that `--t1w-image` is the main switch, and `--t1` is the compatibility switch (same for t1brain and t2).  We should also check that the Batch uses the `t1w` variants.

Reply:
We should also probably edit `Examples/Scripts/FreeSurferPipelineBatch.sh` to use the `--t1w-image`-like argument names.